Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM golang:1.23-alpine3.21 AS builder
+
+RUN apk add --no-cache bash git gcc musl-dev
+
+WORKDIR /src
+COPY . .
+
+RUN . ./scripts/version
+RUN --mount=type=cache,id=gomod,target=/go/pkg/mod \
+    --mount=type=cache,id=gobuild,target=/root/.cache/go-build \
+    CGO_ENABLED=0 go build -ldflags "-X main.VERSION=$VERSION -extldflags -static -s" -o /bin/helm-controller
```

Comment:
1. I don't think this will properly set $VERSION for the second RUN command
2. I don't think you can use `docker run` flags in RUN Dockerfile directives, can you?

Reply:
We don't use `docker run` to produce the binary, its all done by a `docker build` call with an `--output` target

Reply:
right that's what I'm saying. `RUN` commands don't accept `docker run` flags like `--mount`.

Reply:
They do, see https://docs.docker.com/build/cache/optimize/#use-bind-mounts

Reply:
This is the better link, they have a golang cache example even https://docs.docker.com/build/cache/optimize/#use-cache-mounts

Reply:
I still think that VERSION from running the version script won't get set properly here; it needs to go in the same RUN command.

Reply:
Yes that is still a problem, I need to resolve.